filewatcher/filewatcher.go

```diff
@@ -64,6 +66,8 @@ type Result struct {
 
 	ctx    context.Context
 	cancel context.CancelFunc
+	lock   sync.Mutex
```

Comment:
style nit: group (by adding blank lines) lock and the fields it guards together.